From 11f8048dfed84f48f4269ce1ad85e9605e4108b4 Mon Sep 17 00:00:00 2001
From: RubenKelevra <ruben@vfn-nrw.de>
Date: Sat, 14 May 2016 17:48:19 +0200
Subject: [PATCH 1/1] backport: include/image.mk - Add TP-Link image
 safe-loader define

Signed-off-by: Josh Bendavid <joshbendavid@gmail.com>
[darryl@sokoloski.ca: submitting patches on behalf of Josh Bendavid]
Signed-off-by: Darryl Sokoloski <darryl@sokoloski.ca>

source: https://patchwork.ozlabs.org/patch/612485/
---
 include/image.mk | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/include/image.mk b/include/image.mk
index a92b13e..77c1137 100644
--- a/include/image.mk
+++ b/include/image.mk
@@ -282,6 +282,17 @@ define Build/uImage
 	@mv $@.new $@
 endef
 
+define Build/tplink-safe
+	$(STAGING_DIR_HOST)/bin/tplink-safeloader \
+		-B $(TPLINK_BOARD_ID) -V OpenWrt.$(REVISION) \
+		-k $(word 1,$^) \
+		-r $(word 2,$^) \
+		-j \
+		$(if $(findstring sysupgrade,$1),-S) \
+		-o $@.new
+	mv $@.new $@
+endef
+
 define Build/lzma
 	$(STAGING_DIR_HOST)/bin/lzma e $@ -lc1 -lp2 -pb2 $(1) $@.new
 	@mv $@.new $@
-- 
2.8.2

